source/common/http/filter_manager.h

```diff
@@ -840,6 +864,45 @@ class FilterManager : public ScopeTrackedObject,
   void onDownstreamReset() { state_.saw_downstream_reset_ = true; }
 
 private:
+  class FilterChainFactoryCallbacksImpl : public Http::FilterChainFactoryCallbacks {
+  public:
+    FilterChainFactoryCallbacksImpl(FilterManager& manager,
+                                    const Http::FilterFacotryContext& context)
+        : manager_(manager), context_(context) {}
+
+    void addStreamDecoderFilter(Http::StreamDecoderFilterSharedPtr filter) override {
+      manager_.addStreamFilterBase(filter.get());
+      manager_.addStreamDecoderFilter(std::make_unique<ActiveStreamDecoderFilter>(
+          manager_, std::move(filter), false, context_.custom_name, context_.filter_name));
+    }
+
+    void addStreamEncoderFilter(Http::StreamEncoderFilterSharedPtr filter) override {
+      manager_.addStreamFilterBase(filter.get());
+      manager_.addStreamEncoderFilter(std::make_unique<ActiveStreamEncoderFilter>(
+          manager_, std::move(filter), false, context_.custom_name, context_.filter_name));
+    }
+
+    void addStreamFilter(Http::StreamFilterSharedPtr filter) override {
+      StreamDecoderFilter* decoder_filter = filter.get();
+      manager_.addStreamFilterBase(decoder_filter);
+
+      manager_.addStreamDecoderFilter(std::make_unique<ActiveStreamDecoderFilter>(
+          manager_, filter, true, context_.custom_name, context_.filter_name));
+      manager_.addStreamEncoderFilter(std::make_unique<ActiveStreamEncoderFilter>(
+          manager_, std::move(filter), true, context_.custom_name, context_.filter_name));
```

Comment:
nit, since it is shared ptr, so actually, we needn't the std::move here, right?

Reply:
This move is used to avoid unnecessry atomic counter updates.